### PR TITLE
Enable loadbalancer.sticky for ECS

### LINKED ISSUE #1522: Support Session Stickiness in ECS
### Do you want to request a *feature* or report a *bug*?

Feature Request

### What did you do?

Attempted to use `traefik.backend.loadbalancer.sticky=true` label on an ECS service but it was not honored.

### What did you expect to see?

Traefik backend would set sticky affinity and route traffic to the same container for that client.

### What did you see instead?

Traefik routed requests without honoring session stickiness.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.2.3
Codename:     morbier
Go version:   go1.7.5
Built:        2017-04-13_07:32:59PM
OS/Arch:      darwin/amd64
```

### PR DESCRIPTION
**Pull request by mmatur**
### Description

The goal of this PR is to add support of sticky session for ECS.
Low balancer configuration will be always present with default values:
`traefik.backend.loadbalancer.method=wrr`
`traefik.backend.loadbalancer.sticky=false`

To override these values you just to add labels 

Fixes #1522

### CODE REVIEW

**Comment by ldez** [comment]
Could you fix the build?

> Errors from golint:
> provider/ecs/ecs.go:435:1: exported method Provider.LoadBalancerSticky should have comment or be unexported
> provider/ecs/ecs.go:444:1: exported method Provider.LoadBalancerMethod should have comment or be unexported
> 
> Please fix the above errors. You can test via "golint" and commit the result.

**Comment by mmatur** [comment]
This PR is the implementation for feature requested on #1522

File: provider/ecs/ecs.go [hunk 1]
```diff
@@ -420,6 +432,24 @@ func (p *Provider) getFrontendRule(i ecsInstance) string {
 	return "Host:" + strings.ToLower(strings.Replace(i.Name, "_", "-", -1)) + "." + p.Domain
 }
 
+func (p *Provider) getLoadBalancerSticky(instances []ecsInstance) string {
+	for instances != nil && len(instances) > 0 {
+		if label := instances[0].label(types.LabelBackendLoadbalancerSticky); label != "" {
```
Comment (ldez) [inline provider/ecs/ecs.go]: could you split in two lines?

File: provider/ecs/ecs.go [hunk 2]
```diff
@@ -420,6 +432,24 @@ func (p *Provider) getFrontendRule(i ecsInstance) string {
...
+			return label
+		}
+	}
+	return "false"
+}
+
+func (p *Provider) getLoadBalancerMethod(instances []ecsInstance) string {
+	for instances != nil && len(instances) > 0 {
+		if label := instances[0].label(types.LabelBackendLoadbalancerMethod); label != "" {
```
Comment (ldez) [inline provider/ecs/ecs.go]: could you split in two lines?

File: provider/ecs/ecs.go [hunk 3]
```diff
@@ -420,6 +432,24 @@ func (p *Provider) getFrontendRule(i ecsInstance) string {
 	return "Host:" + strings.ToLower(strings.Replace(i.Name, "_", "-", -1)) + "." + p.Domain
 }
 
+func (p *Provider) getLoadBalancerSticky(instances []ecsInstance) string {
+	for instances != nil && len(instances) > 0 {
```
Comment (ldez) [inline provider/ecs/ecs.go]: `instances != nil` is unnecessary.

```go
func ExampleEmptySlice() {
	var foo []string
	fii := bar()
	fmt.Println(foo, fii)
	// output: [] []
}
func bar() []string{
	return nil
}
```

File: provider/ecs/ecs.go [hunk 4]
```diff
@@ -420,6 +432,24 @@ func (p *Provider) getFrontendRule(i ecsInstance) string {
...
+		if label := instances[0].label(types.LabelBackendLoadbalancerSticky); label != "" {
+			return label
+		}
+	}
+	return "false"
+}
+
+func (p *Provider) getLoadBalancerMethod(instances []ecsInstance) string {
+	for instances != nil && len(instances) > 0 {
```
Comment (ldez) [inline provider/ecs/ecs.go]: `instances != nil` is unnecessary.

```go
func ExampleEmptySlice() {
	var foo []string
	fii := bar()
	fmt.Println(foo, fii)
	// output: [] []
}
func bar() []string{
	return nil
}
```

File: provider/ecs/ecs.go [hunk 5]
```diff
@@ -189,10 +191,20 @@ func (p *Provider) loadECSConfig(ctx context.Context, client *awsClient) (*types
 
 	instances = fun.Filter(p.filterInstance, instances).([]ecsInstance)
 
+	services := make(map[string][]ecsInstance)
+
+	for _, i := range instances {
```
Comment (ldez) [inline provider/ecs/ecs.go]: could you rename `i` to `instance` ?

I know it's already use but I think `i` is an ambiguous name in a `for`.

File: provider/ecs/ecs.go (hunk 4 above)
Comment (mmatur) [inline provider/ecs/ecs.go]: Thanks for the go course. I will fix that asap

**Pull request merged**